src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/participationhistory/domain/CourseParticipation.kt

```diff
@@ -77,27 +76,21 @@ data class CourseParticipationSetting(
   var location: String? = null,
 
   @Enumerated(EnumType.STRING)
-  var type: CourseSetting?,
-
-  @Formula("0")
-  private val ignoreMe: Int = 0, // This unused, non-nullable field forces Hibernate to create an @Embedded instance when all fields are null.
+  var type: CourseSetting,
 )
 
 @Embeddable
 data class CourseOutcome(
   @Enumerated(EnumType.STRING)
   @Column(name = "outcome_status")
-  var status: CourseStatus? = null,
+  var status: CourseStatus,
```

Comment:
This is probably for a separate PR, but would it make sense to namespace the `CourseSetting` and `CourseStatus` enums more explicitly, e.g. `CourseParticipationSettingType` and `CourseParticipationOutcomeStatus`?

They're both about course participations, not courses, and having a `Setting` be a property of a `Setting` is maybe a little confusing?

Reply:
Agreed. I'd actually prefer to simplify this down even further - i.e. have a flat structure across the entire entity, and just shift things up a layer. But I don't mind sticking to making name changes here / in a follow-up PR for now.

Reply:
I'll implement the name changes in an immediate follow-up PR in exchange for an approval now 😁

Reply:
This should also be a `CourseParticipationOutcome` I guess, but if you're flattening that might be irrelevant